src/common/Icons.js

```diff
@@ -44,12 +44,12 @@ export type IconNames = FeatherGlyphs;
 export const Icon = fixIconType<IconNames>(Feather);
 
 /** A (type for a) component-type like `Icon` but with `name` already specified. */
-export type IconType = ComponentType<$Diff<IconProps<empty>, {| name: mixed |}>>;
+export type SpecificIconType = ComponentType<$Diff<IconProps<empty>, {| name: mixed |}>>;
```

Comment:
I read this as a specific (icon-type), rather than a (specific icon)'s type.

I don't think it improves legibility at any use site.

Reply:
> I read this as a specific (icon-type), rather than a (specific icon)'s type.

You've read it correctly!

This name can probably be improved. The thing that's really wrong with the existing names is the relationship between `Icon` and this `IconType`: with those two names next to each other in the same small piece of API, it sure looks like `IconType` should be the type of `Icon`, but it's not... and in fact the difference between them runs in a completely different direction from that.

One question here is: why is "Type" in the name at all? Yes this is the name of a type... but if we had a convention that types' names said "Type", we'd have a lot more names like that. I'm taking "Type" here as parallel to the relationship in React (as described with Flow) between `Component` and `ComponentType`. One of those is a class -- so already a type -- and the other is the type *of* a thing like `Component`.

So `IconType` would be a natural name for the type of `Icon` and things like it... and this isn't that, but instead is a certain variation on that. "Specific" is one attempt to gesture at that variation.

Reply:
Yeah, renaming to `FeatherIcon` might be a good further cleanup. That would also help with the awkward mixture we have of places where `Icon` means
* sometimes a value like `IconDone` or `IconTerminal` or `IconPrivateChat`, of type `SpecificIconType`;
* other times the `Icon` export of this module, which has a different type -- it requires a `name` prop.

Precisely because it's an awkward mixture it's a little annoying to go in and fix, though. So I'm going to go ahead and merge this without blocking on that additional fixup.